### PR TITLE
ISSUE-220: replace using npx to execute action-validator, use the binary instead

### PR DESCRIPTION
**Pull request by beto-aveiga**
### Description
- Issue: #220 
- Dowloads the binary of action-validator instead of using `npx`. Based on https://github.com/Lullabot/drainpipe/issues/220#issuecomment-1614915343
  - It took about 2 seconds to download the binary on my computer.
  - I tested that `action-validator` was executable with a new container, like: 
    - `docker run -it tugboatqa/php:8.2.15-apache-bullseye /bin/bash`

### CODE REVIEW

**Review by justafish — CHANGES_REQUESTED** [review]
Thanks for the PR. This would download the binary each time the task is run, instead could you please change it to use the the same installation method as we use for task and the local security checker e.g.
https://github.com/Lullabot/drainpipe/blob/main/src/BinaryInstaller.php
https://github.com/Lullabot/drainpipe/blob/main/drainpipe-dev/src/DevBinaryInstallerPlugin.php

**Comment by beto-aveiga** [comment]
@justafish I noticed that `action-validator` is only available for
- linux amd64
- linux arm64
- darwin amd64
- darwin arm64

Is that something we need to worry about? Or do something different in the binary installer? Or fallback to `npx` when the binary is not found?

**Comment by beto-aveiga** [comment]
I just added a fallback in case the binary is not available.

**Comment by deviantintegral** [comment]
> I just added a fallback in case the binary is not available.

That would be if you were running on Windows, directly on the host? Do we actually support doing that? @justafish thoughts?

If we keep it, I think we should document that it's specifically for Windows support as a comment.

File: tasks/github.yml [hunk 1]
```diff
@@ -13,7 +13,13 @@ tasks:
...
           directories=".github/workflows .github/actions"
           if [ "{{ .directories }}" != "" ]; then directories="{{ .directories }}"; fi
-          find $directories -type f \( -iname '*.yaml' -o -iname '*.yml' \)  | xargs -n1 npx @action-validator/cli
+
+          if command -v action-validator &> /dev/null
+          then
+              find $directories -type f \( -iname '*.yaml' -o -iname '*.yml' \) | xargs -n1 action-validator
+          else
+              find $directories -type f \( -iname '*.yaml' -o -iname '*.yml' \) | xargs -n1 npx @action-validator/cli
+          fi
```
Comment (justafish) [inline tasks/github.yml:22]:
```suggestion
          if [ -f ./vendor/bin/action-validator ]; then
              find $directories -type f \( -iname '*.yaml' -o -iname '*.yml' \) | xargs -n1 ./vendor/bin/action-validator
          else
              find $directories -type f \( -iname '*.yaml' -o -iname '*.yml' \) | xargs -n1 npx @action-validator/cli
          fi
```

File: tasks/github.yml [hunk 2]
```diff
@@ -13,7 +13,13 @@ tasks:
       - |
           directories=".github/workflows .github/actions"
           if [ "{{ .directories }}" != "" ]; then directories="{{ .directories }}"; fi
-          find $directories -type f \( -iname '*.yaml' -o -iname '*.yml' \)  | xargs -n1 npx @action-validator/cli
+
+          if [ -f ./vendor/bin/action-validator ]; then
+              find $directories -type f \( -iname '*.yaml' -o -iname '*.yml' \) | xargs -n1 ./vendor/bin/action-validator
+          # Fallback for architectures that don't have a release e.g. Windows
+          else
```
Comment (justafish) [inline tasks/github.yml:20]:
```suggestion
          # Fallback for architectures that don't have a release e.g. Windows
          else
```

File: tasks/github.yml [hunk 3]
```diff
@@ -13,7 +13,14 @@ tasks:
       - |
           directories=".github/workflows .github/actions"
           if [ "{{ .directories }}" != "" ]; then directories="{{ .directories }}"; fi
-          find $directories -type f \( -iname '*.yaml' -o -iname '*.yml' \)  | xargs -n1 npx @action-validator/cli
+
+          if [ -f ./vendor/bin/action-validator ]; then
+              find $directories -type f \( -iname '*.yaml' -o -iname '*.yml' \) | xargs -n1 ./vendor/bin/action-validator
+          # Fallback for architectures that don't have a release e.g. Windows
+          # Fallback for architectures that don't have a release e.g. Windows
```
Comment (justafish) [inline tasks/github.yml:20]:
```suggestion
```

**Review by justafish — APPROVED** [review]
I added a minor tweak to the explicit path of the binary, and added Andrew's suggestion for a comment. I've tested this locally with the drainpipe-test script and all looks to be working well :+1: 

Thanks all!

**Pull request merged**